File: AtlasWeb/Core/Modules/Tibbi_Surec/Hasta_Gecmisi/SpacialityModules/OldWomanSpecialityFormViewModel.ts
//$92D04909
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { WomanSpecialityObject } from 'NebulaClient/Model/AtlasClientModel';
import { PregnancyComplications } from 'NebulaClient/Model/AtlasClientModel';
import { ObligedRiskFactors } from 'NebulaClient/Model/AtlasClientModel';
import { PregnancyDangerSign } from 'NebulaClient/Model/AtlasClientModel';
import { FetusFollow } from 'NebulaClient/Model/AtlasClientModel';
import { PregnancyFollow } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSGebelikteRiskFaktorleri } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSGebelikBildirimiZorunluRiskFaktorleri } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSGebelikLohusalikSeyrindeTehlikeIsareti } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSKacinciGebeIzlem } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSKadinSagligiIslemleri } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSKonjenitalAnomaliliDogumVarligi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSIdrardaProtein } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSDVitaminiLojistigiveDestegi } from 'NebulaClient/Model/AtlasClientModel';
import { SKRSDemirLojistigiveDestegi } from 'NebulaClient/Model/AtlasClientModel';
import { Infertility } from 'NebulaClient/Model/AtlasClientModel';
import { Gynecology } from 'NebulaClient/Model/AtlasClientModel';

import { Type } from 'NebulaClient/ClassTransformer';

import { InfertilityPatientInformation } from 'NebulaClient/Model/AtlasClientModel';

export class OldWomanSpecialityFormViewModel extends BaseViewModel {
    protected __type__: string = 'Core.Models.OldWomanSpecialityFormViewModel, Core';

    @Type(() => WomanSpecialityObject)
    public _WomanSpecialityObject: WomanSpecialityObject = new WomanSpecialityObject();

    @Type(() => PregnancyComplications)
    public PregnancyComplicationsGridList: Array<PregnancyComplications> = new Array<PregnancyComplications>();

    @Type(() => ObligedRiskFactors)
    public ObligedRiskFactorsGridList: Array<ObligedRiskFactors> = new Array<ObligedRiskFactors>();

    @Type(() => PregnancyDangerSign)
    public PregnancyDangerSignGridList: Array<PregnancyDangerSign> = new Array<PregnancyDangerSign>();

    @Type(() => FetusFollow)
    public FetusFollowGridList: Array<FetusFollow> = new Array<FetusFollow>();

    @Type(() => PregnancyFollow)
    public PregnancyFollows: Array<PregnancyFollow> = new Array<PregnancyFollow>();
    public SKRSGebelikteRiskFaktorleris: Array<SKRSGebelikteRiskFaktorleri> = new Array<SKRSGebelikteRiskFaktorleri>();
    public SKRSGebelikBildirimiZorunluRiskFaktorleris: Array<SKRSGebelikBildirimiZorunluRiskFaktorleri> = new Array<SKRSGebelikBildirimiZorunluRiskFaktorleri>();
    public SKRSGebelikLohusalikSeyrindeTehlikeIsaretis: Array<SKRSGebelikLohusalikSeyrindeTehlikeIsareti> = new Array<SKRSGebelikLohusalikSeyrindeTehlikeIsareti>();
    public SKRSKacinciGebeIzlems: Array<SKRSKacinciGebeIzlem> = new Array<SKRSKacinciGebeIzlem>();
    public SKRSKadinSagligiIslemleris: Array<SKRSKadinSagligiIslemleri> = new Array<SKRSKadinSagligiIslemleri>();
    public SKRSKonjenitalAnomaliliDogumVarligis: Array<SKRSKonjenitalAnomaliliDogumVarligi> = new Array<SKRSKonjenitalAnomaliliDogumVarligi>();
    public SKRSIdrardaProteins: Array<SKRSIdrardaProtein> = new Array<SKRSIdrardaProtein>();
    public SKRSDVitaminiLojistigiveDestegis: Array<SKRSDVitaminiLojistigiveDestegi> = new Array<SKRSDVitaminiLojistigiveDestegi>();
    public SKRSDemirLojistigiveDestegis: Array<SKRSDemirLojistigiveDestegi> = new Array<SKRSDemirLojistigiveDestegi>();

    @Type(() => Infertility)
    public Infertilitys: Array<Infertility> = new Array<Infertility>();

    @Type(() => Gynecology)
    public Gynecologys: Array<Gynecology> = new Array<Gynecology>();

    @Type(() => InfertilityPatientInformation)
    public _InfertilityPatientInformation: InfertilityPatientInformation = new InfertilityPatientInformation();

    public BloodGroup: string;
    public HusbandBloodGroup: string;
    public DegreeOfRelationship: string;
    public RhIncompatibility: string;
    public PreviousBirthControlMethod: string;
    public CurrentBirthControlMethod: string;
    public GenitalAbnormalities: string;
    public MenstrualCycleAbnormalities: string;
    public ReproductiveAbnormality: string;
    public Dysuria: string;
    public Dyspareunia: string;
    public Hirsutism: string;
    public VaginalDischarge: string;
    public PregnancyDiseases: string;
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Hasta_Gecmisi/SpacialityModules/OldWomanSpecialityFormViewModel.cs
//$73C579A6
using System;
using System.Linq;
using TTInstanceManagement;
using Core.Models;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using System.Text;

namespace Core.Controllers
{
    public partial class WomanSpecialityObjectServiceController
    {
        partial void PreScript_OldWomanSpecialityForm(OldWomanSpecialityFormViewModel viewModel, WomanSpecialityObject womanSpecialityObject, TTObjectContext objectContext)
        {
            viewModel.BloodGroup = womanSpecialityObject.PhysicianApplication.Episode.Patient.BloodGroupType != null ? womanSpecialityObject.PhysicianApplication.Episode.Patient.BloodGroupType.KODU : "";
            viewModel.HusbandBloodGroup = womanSpecialityObject.HusbandBloodGroup != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.HusbandBloodGroup) : "";
            viewModel.DegreeOfRelationship = womanSpecialityObject.DegreeOfRelationship != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.DegreeOfRelationship) : "";
            viewModel.RhIncompatibility = womanSpecialityObject.RhIncompatibility != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.RhIncompatibility) : "";
            //Jinekoloji
            viewModel.PreviousBirthControlMethod = womanSpecialityObject.Gynecology.PreviousBirthControlMethod != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.Gynecology.PreviousBirthControlMethod) : "";
            viewModel.CurrentBirthControlMethod = womanSpecialityObject.Gynecology.CurrentBirthControlMethod != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.Gynecology.CurrentBirthControlMethod) : "";
            viewModel.GenitalAbnormalities = womanSpecialityObject.Gynecology.GenitalAbnormalities != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.Gynecology.GenitalAbnormalities) : "";
            viewModel.MenstrualCycleAbnormalities = womanSpecialityObject.Gynecology.MenstrualCycleAbnormalities != null ? Common.GetDisplayTextOfDataTypeEnum(womanSpecialityObject.Gynecology.MenstrualCycleAbnormalities) : "";
            viewModel.ReproductiveAbnormality = womanSpecialityObject.Gynecology.ReproductiveAbnormality != null ? womanSpecialityObject.Gynecology.ReproductiveAbnormality.Name : "";
            viewModel.Dysuria = womanSpecialityObject.Gynecology.Dysuria == true ? "Var" : TTUtils.CultureService.GetText("M24703", "Yok");
            viewModel.Dyspareunia = womanSpecialityObject.Gynecology.Dyspareunia == true ? "Var" : TTUtils.CultureService.GetText("M24703", "Yok");
            viewModel.Hirsutism = womanSpecialityObject.Gynecology.Hirsutism == true ? "Var" : TTUtils.CultureService.GetText("M24703", "Yok");
            viewModel.VaginalDischarge = womanSpecialityObject.Gynecology.VaginalDischarge == true ? "Var" : TTUtils.CultureService.GetText("M24703", "Yok");
            //Hasta Bazlı İnfertilite
            if (womanSpecialityObject.PhysicianApplication.Episode.Patient.InfertilityPatientInformation != null)
            {
                viewModel._InfertilityPatientInformation = womanSpecialityObject.PhysicianApplication.Episode.Patient.InfertilityPatientInformation;
            }

            if (viewModel._WomanSpecialityObject.PregnancyFollow != null)
            {
                StringBuilder str = new StringBuilder();
                str.Append(viewModel._WomanSpecialityObject.PregnancyFollow.Tension == true ? "Tansiyon, " : "");
                str.Append(viewModel._WomanSpecialityObject.PregnancyFollow.GestationalDiabetes == true ? "Gebelik Şekeri, " : "");
                str.Append(viewModel._WomanSpecialityObject.PregnancyFollow.CardiovascularDiseases == true ? "Damar Hastalıkları, " : "");
                str.Append(viewModel._WomanSpecialityObject.PregnancyFollow.Anemia == true ? "Anemi, " : "");
                str.Append(viewModel._WomanSpecialityObject.PregnancyFollow.Nausea == true ? "Bulantı, " : "");
                str.Append(viewModel._WomanSpecialityObject.PregnancyFollow.Bleeding == true ? "Kanama, " : "");
                viewModel.PregnancyDiseases = str.ToString();
            }
        }
    }
}

namespace Core.Models
{
    public partial class OldWomanSpecialityFormViewModel
    {
        public string BloodGroup
        {
            get;
            set;
        }
        public string HusbandBloodGroup { get; set; }
        public string DegreeOfRelationship { get; set; }
        public string RhIncompatibility { get; set; }

        //Jinekoloji
        public string PreviousBirthControlMethod
        {
            get;
            set;
        }

        public string CurrentBirthControlMethod
        {
            get;
            set;
        }

        public string GenitalAbnormalities
        {
            get;
            set;
        }

        public string MenstrualCycleAbnormalities
        {
            get;
            set;
        }

        public string ReproductiveAbnormality
        {
            get;
            set;
        }

        public string Dysuria
        {
            get;
            set;
        }

        public string Dyspareunia
        {
            get;
            set;
        }

        public string Hirsutism
        {
            get;
            set;
        }

        public string VaginalDischarge
        {
            get;
            set;
        }

        //Hasta Bazlı İnfertilite
        public TTObjectClasses.InfertilityPatientInformation _InfertilityPatientInformation
        {
            get;
            set;
        }

        //Gebe İzlem
        public string PregnancyDiseases
        {
            get;
            set;
        }
    }
}